src/segment/event.rs

```diff
@@ -50,6 +50,7 @@ pub(crate) struct PendingEvent {
     pub(crate) data: Vec<u8>,
     pub(crate) conditional_offset: Option<i64>,
     pub(crate) oneshot_sender: oneshot::Sender<Result<(), Error>>,
+    pub(crate) flush_oneshot_sender: Option<oneshot::Sender<Result<(), Error>>>,
```

Comment:
flush_oneshot_sender is never used anywhere; we would need to invoke a send() with the write status on it.